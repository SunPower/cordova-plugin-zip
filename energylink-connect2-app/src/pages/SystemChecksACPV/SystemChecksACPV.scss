@import '~@sunpower/theme-dark/styles/variables';

.system-checks {
  display: grid;
  grid-template-rows: 20px 1fr 135px;
  height: calc(100vh - 235px);

  .is-text {
    text-decoration: none;
    &:hover {
      background-color: transparent;
    }
    &:focus {
      box-shadow: none;
    }
  }

  .simple-error {
    padding: 5px 10px;
    text-align: left;
    background-color: $silver-sand4;
    border-radius: 5px;
  }
}
