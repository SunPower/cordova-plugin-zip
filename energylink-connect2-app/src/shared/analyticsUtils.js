import {
  __,
  compose,
  curry,
  gt,
  isEmpty,
  join,
  lensIndex,
  map,
  not,
  over,
  props,
  propSatisfies,
  toLower,
  toUpper,
  when
} from 'ramda'

export const isNotEmpty = compose(not, isEmpty)

export const capitalizeWord = when(
  propSatisfies(gt(__, 0), 'length'),
  compose(join(''), over(lensIndex(0), toUpper), toLower)
)

const stringValueOrEmptyString = stringValue => stringValue || ''

const getStringProps = curry(compose(map(stringValueOrEmptyString), props))

export const getUserProfile = getStringProps([
  'uniqueId',
  'firstName',
  'lastName',
  'email',
  'dealerName',
  'recordType'
])
