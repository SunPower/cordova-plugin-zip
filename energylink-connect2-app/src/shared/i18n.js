import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
