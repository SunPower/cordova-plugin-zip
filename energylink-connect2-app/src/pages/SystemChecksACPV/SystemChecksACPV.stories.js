import { storiesOf } from '@storybook/react'
import { always, evolve } from 'ramda'
import React from 'react'
import { Provider } from 'react-redux'

import SystemChecksACPV from '.'

import { SYSTEM_CHECKS_STATUS } from 'state/reducers/systemChecks'
import { configureStore } from 'state/store'

export const systemChecksDefaultState = {
  pvs: {
    fwVersion: 60002
  },
  systemChecks: {
    productionCTProgress: 0,
    productionCTErrors: [],
    productionCTStatus: SYSTEM_CHECKS_STATUS.NOT_RUNNING,
    consumptionCTProgress: 0,
    consumptionCTErrors: [],
    consumptionCTStatus: SYSTEM_CHECKS_STATUS.NOT_RUNNING,
    overallStatus: SYSTEM_CHECKS_STATUS.NOT_RUNNING,
    overallErrors: []
  }
}

export const systemChecksDefaultStateRUNNING = evolve(
  {
    pvs: {
      fwVersion: 60002
    },
    systemChecks: {
      productionCTProgress: always(10),
      productionCTStatus: always(SYSTEM_CHECKS_STATUS.RUNNING),
      consumptionCTStatus: always(SYSTEM_CHECKS_STATUS.RUNNING),
      consumptionCTErrors: always([])
    }
  },
  systemChecksDefaultState
)

export const systemChecksDefaultStateFAILED = evolve(
  {
    pvs: {
      fwVersion: 60002
    },
    systemChecks: {
      productionCTProgress: always(100),
      productionCTErrors: always([]),
      productionCTStatus: always(SYSTEM_CHECKS_STATUS.FAILED),
      consumptionCTProgress: always(100),
      consumptionCTStatus: always(SYSTEM_CHECKS_STATUS.SUCCEEDED),
      overallStatus: always(SYSTEM_CHECKS_STATUS.FAILED),
      overallErrors: always([{ error_code: '33024' }])
    }
  },
  systemChecksDefaultState
)

export const systemChecksDefaultStateSUCCEEDED = evolve(
  {
    pvs: {
      fwVersion: 60002
    },
    systemChecks: {
      productionCTProgress: always(100),
      productionCTStatus: always(SYSTEM_CHECKS_STATUS.SUCCEEDED),
      consumptionCTProgress: always(100),
      consumptionCTStatus: always(SYSTEM_CHECKS_STATUS.SUCCEEDED),
      overallStatus: always(SYSTEM_CHECKS_STATUS.SUCCEEDED)
    }
  },
  systemChecksDefaultState
)

storiesOf('SystemChecksACPV Page', module)
  .add('NOT_RUNNING', () => {
    const { store } = configureStore(systemChecksDefaultState)

    return (
      <div className="full-min-height pl-10 pr-10">
        <Provider store={store}>
          <SystemChecksACPV />
        </Provider>
      </div>
    )
  })
  .add('RUNNING', () => {
    const { store } = configureStore(systemChecksDefaultStateRUNNING)

    return (
      <div className="full-min-height pl-10 pr-10">
        <Provider store={store}>
          <SystemChecksACPV />
        </Provider>
      </div>
    )
  })
  .add('FAILED', () => {
    const { store } = configureStore(systemChecksDefaultStateFAILED)

    return (
      <div className="full-min-height pl-10 pr-10">
        <Provider store={store}>
          <SystemChecksACPV />
        </Provider>
      </div>
    )
  })
  .add('SUCCEEDED', () => {
    const { store } = configureStore(systemChecksDefaultStateSUCCEEDED)

    return (
      <div className="full-min-height pl-10 pr-10">
        <Provider store={store}>
          <SystemChecksACPV />
        </Provider>
      </div>
    )
  })
