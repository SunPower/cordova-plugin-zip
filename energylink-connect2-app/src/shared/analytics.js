import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
