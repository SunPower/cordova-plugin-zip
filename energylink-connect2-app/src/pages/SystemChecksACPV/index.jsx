import { pathOr, length } from 'ramda'
import React, { useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'

import { Loader } from 'components/Loader'
import StatusBox from 'components/StatusBox'
import paths from 'routes/paths'
import { useI18n } from 'shared/i18n'
import { capitalize, either, trimWarnings } from 'shared/utils'
import {
  RESET_SYSTEM_CHECKS,
  SYSTEM_CHECKS_INIT
} from 'state/actions/systemChecks'
import { SUBMIT_COMMISSION_INIT } from 'state/actions/systemConfiguration'
import { SYSTEM_CHECKS_STATUS } from 'state/reducers/systemChecks'

import './SystemChecksACPV.scss'

function SystemChecksACPV() {
  const t = useI18n()
  const dispatch = useDispatch()
  const history = useHistory()

  const {
    productionCTProgress,
    productionCTErrors,
    productionCTStatus,
    consumptionCTProgress,
    consumptionCTErrors,
    consumptionCTStatus,
    overallStatus,
    overallErrors
  } = useSelector(pathOr({}, ['systemChecks']))

  const showSummary =
    overallStatus === SYSTEM_CHECKS_STATUS.SUCCEEDED ||
    overallStatus === SYSTEM_CHECKS_STATUS.FAILED

  const pctErrorsLength = length(trimWarnings(productionCTErrors))
  const cctErrorsLength = length(trimWarnings(consumptionCTErrors))
  const overallErrorsLength = length(trimWarnings(overallErrors))
  const sumErrors = pctErrorsLength + cctErrorsLength + overallErrorsLength

  const hasErrors = sumErrors > 0

  const productionCT = {
    text: getText(pctErrorsLength, productionCTStatus, t),
    indicator: getIndicator(productionCTStatus, productionCTProgress),
    title: t('PRODUCTION_CTE')
  }

  const consumptionCT = {
    text: getText(cctErrorsLength, consumptionCTStatus, t),
    indicator: getIndicator(consumptionCTStatus, consumptionCTProgress),
    title: t('CONSUMPTION_CTE')
  }

  const generalErrors = {
    text: getText(overallErrorsLength, overallStatus, t),
    indicator: getIndicator(overallStatus, 100),
    title: t('GENERAL_CTE')
  }

  const isRunning = overallStatus === SYSTEM_CHECKS_STATUS.RUNNING

  useEffect(() => {
    if (overallStatus === SYSTEM_CHECKS_STATUS.NOT_RUNNING)
      dispatch(SYSTEM_CHECKS_INIT())
  }, [dispatch, overallStatus])

  const commission = () => {
    dispatch(SUBMIT_COMMISSION_INIT())
    history.push(paths.PROTECTED.SAVING_CONFIGURATION.path)
  }

  const restartChecks = () => {
    dispatch(RESET_SYSTEM_CHECKS())
    dispatch(SYSTEM_CHECKS_INIT())
  }

  return (
    <main className="page-height system-checks pl-10 pr-10">
      <header className="is-clearfix has-text">
        <button
          className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-left is-size-6"
          disabled={isRunning}
          onClick={history.goBack}
        >
          <i className="sp-chevron-left has-text-primary is-size-5 mr-5 mt-5" />
          {t('SYSTEM_CHECKS')}
        </button>

        <button
          className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-right is-size-6"
          disabled={isRunning}
          onClick={restartChecks}
        >
          <i className="sp-update left has-text-primary is-size-5 mr-5 mt-5" />
          {!isRunning
            ? t('RUN_AGAIN')
            : capitalize(SYSTEM_CHECKS_STATUS.RUNNING)}
        </button>
      </header>

      <section className="mt-20">
        <StatusBox {...productionCT} />
        <StatusBox {...consumptionCT} />

        {either(
          overallErrorsLength > 0 && !isRunning,
          <StatusBox {...generalErrors} />
        )}

        {either(
          hasErrors,
          <p className="has-text-primary is-size-6 has-text-centered">
            {t('SYSTEM_CHECKS_ERRORS', sumErrors)}
          </p>
        )}

        {either(isRunning, <Loader />)}
      </section>

      {either(
        showSummary,
        either(
          hasErrors,

          <section className="mt-10">
            <section className="is-flex is-vertical">
              <button
                className="button is-primary is-outlined is-uppercase mt-20 is-fullwidth mr-5 is-fullwidth"
                onClick={() =>
                  history.push(paths.PROTECTED.SYSTEM_CONFIGURATION.path)
                }
              >
                {t('BACK_TO_SETTINGS')}
              </button>
              <button
                className="button is-primary is-uppercase mt-20 is-fullwidth ml-5 is-fullwidth"
                onClick={() =>
                  history.push(paths.PROTECTED.SYSTEM_CHECKS_ERROR_LIST.path)
                }
              >
                {t('VIEW_ERRORS')}
              </button>
            </section>
            <button
              className="button is-primary is-outlined is-uppercase mt-20 is-fullwidth mr-5"
              onClick={() => history.push(paths.PROTECTED.DATA.path)}
            >
              {t('GO_TO_LIVE_DATA')}
            </button>
          </section>,

          <section className="mt-10">
            <section className="is-flex">
              <button
                className="button is-primary is-outlined is-uppercase mt-20 is-fullwidth mr-5"
                onClick={() => history.push(paths.PROTECTED.DATA.path)}
              >
                {t('GO_TO_LIVE_DATA')}
              </button>
              <button
                className="button is-primary is-uppercase mt-20 is-fullwidth ml-5 is-fullwidth"
                onClick={commission}
              >
                {t('COMMISSION')}
              </button>
            </section>
          </section>
        )
      )}
    </main>
  )
}

const getText = (
  errorsCount = 0,
  status = SYSTEM_CHECKS_STATUS.NOT_RUNNING,
  t
) =>
  either(
    errorsCount > 0,
    <span className="has-text-primary">
      {errorsCount === 1 ? t('ERROR_FOUND') : t('ERRORS_FOUND', errorsCount)}
    </span>,
    capitalize(t(status))
  )

const getIndicator = (status, percent = 0) =>
  either(
    status === SYSTEM_CHECKS_STATUS.FAILED ||
      status === SYSTEM_CHECKS_STATUS.UNSUPPORTED,
    <span className="sp-hey has-text-primary is-size-2" />,
    `${percent}%`
  )

export default SystemChecksACPV
