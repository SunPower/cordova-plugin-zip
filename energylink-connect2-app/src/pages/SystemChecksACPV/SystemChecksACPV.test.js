import React from 'react'
import * as reactRedux from 'react-redux'

import * as i18n from '../../shared/i18n'

import {
  systemChecksDefaultState,
  systemChecksDefaultStateFAILED,
  systemChecksDefaultStateRUNNING,
  systemChecksDefaultStateSUCCEEDED
} from './SystemChecksACPV.stories'

import SystemChecksACPV from '.'

describe('SystemChecksACPV component', () => {
  let dispatchMock

  beforeEach(() => {
    dispatchMock = jest.fn()
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => dispatchMock)
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(() => (key, data) => (data ? `${data} ${key}` : key))
  })

  test('render correctly', () => {
    const { component } = mountWithProvider(<SystemChecksACPV />)(
      systemChecksDefaultState
    )
    expect(component).toMatchSnapshot()
  })

  test('render correctly when RUNNING', () => {
    const { component } = mountWithProvider(<SystemChecksACPV />)(
      systemChecksDefaultStateRUNNING
    )
    expect(component).toMatchSnapshot()
  })

  test('render correctly when FAILED', () => {
    const { component } = mountWithProvider(<SystemChecksACPV />)(
      systemChecksDefaultStateFAILED
    )
    expect(component.find('section button').length).toBe(3)
    expect(component).toMatchSnapshot()
  })

  test('render correctly when SUCCEEDED', () => {
    const { component } = mountWithProvider(<SystemChecksACPV />)(
      systemChecksDefaultStateSUCCEEDED
    )
    expect(component.find('section button').length).toBe(2)
    expect(component).toMatchSnapshot()
  })
})
