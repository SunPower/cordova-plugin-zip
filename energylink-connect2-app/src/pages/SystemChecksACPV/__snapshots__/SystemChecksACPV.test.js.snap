// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemChecksACPV component render correctly 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <SystemChecksACPV>
        <main
          className="page-height system-checks pl-10 pr-10"
        >
          <header
            className="is-clearfix has-text"
          >
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-left is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-chevron-left has-text-primary is-size-5 mr-5 mt-5"
              />
              SYSTEM_CHECKS
            </button>
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-right is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-update left has-text-primary is-size-5 mr-5 mt-5"
              />
              RUN_AGAIN
            </button>
          </header>
          <section
            className="mt-20"
          >
            <StatusBox
              indicator="0%"
              text="Not_running"
              title="PRODUCTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    PRODUCTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Not_running
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  0%
                </article>
              </section>
            </StatusBox>
            <StatusBox
              indicator="0%"
              text="Not_running"
              title="CONSUMPTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    CONSUMPTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Not_running
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  0%
                </article>
              </section>
            </StatusBox>
          </section>
        </main>
      </SystemChecksACPV>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`SystemChecksACPV component render correctly when FAILED 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <SystemChecksACPV>
        <main
          className="page-height system-checks pl-10 pr-10"
        >
          <header
            className="is-clearfix has-text"
          >
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-left is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-chevron-left has-text-primary is-size-5 mr-5 mt-5"
              />
              SYSTEM_CHECKS
            </button>
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-right is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-update left has-text-primary is-size-5 mr-5 mt-5"
              />
              RUN_AGAIN
            </button>
          </header>
          <section
            className="mt-20"
          >
            <StatusBox
              indicator={
                <span
                  className="sp-hey has-text-primary is-size-2"
                />
              }
              text="Failed"
              title="PRODUCTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    PRODUCTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Failed
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  <span
                    className="sp-hey has-text-primary is-size-2"
                  />
                </article>
              </section>
            </StatusBox>
            <StatusBox
              indicator="100%"
              text="Succeeded"
              title="CONSUMPTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    CONSUMPTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Succeeded
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  100%
                </article>
              </section>
            </StatusBox>
            <StatusBox
              indicator={
                <span
                  className="sp-hey has-text-primary is-size-2"
                />
              }
              text={
                <span
                  className="has-text-primary"
                >
                  ERROR_FOUND
                </span>
              }
              title="GENERAL_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    GENERAL_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    <span
                      className="has-text-primary"
                    >
                      ERROR_FOUND
                    </span>
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  <span
                    className="sp-hey has-text-primary is-size-2"
                  />
                </article>
              </section>
            </StatusBox>
            <p
              className="has-text-primary is-size-6 has-text-centered"
            >
              1 SYSTEM_CHECKS_ERRORS
            </p>
          </section>
          <section
            className="mt-10"
          >
            <section
              className="is-flex is-vertical"
            >
              <button
                className="button is-primary is-outlined is-uppercase mt-20 is-fullwidth mr-5 is-fullwidth"
                onClick={[Function]}
              >
                BACK_TO_SETTINGS
              </button>
              <button
                className="button is-primary is-uppercase mt-20 is-fullwidth ml-5 is-fullwidth"
                onClick={[Function]}
              >
                VIEW_ERRORS
              </button>
            </section>
            <button
              className="button is-primary is-outlined is-uppercase mt-20 is-fullwidth mr-5"
              onClick={[Function]}
            >
              GO_TO_LIVE_DATA
            </button>
          </section>
        </main>
      </SystemChecksACPV>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`SystemChecksACPV component render correctly when RUNNING 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <SystemChecksACPV>
        <main
          className="page-height system-checks pl-10 pr-10"
        >
          <header
            className="is-clearfix has-text"
          >
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-left is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-chevron-left has-text-primary is-size-5 mr-5 mt-5"
              />
              SYSTEM_CHECKS
            </button>
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-right is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-update left has-text-primary is-size-5 mr-5 mt-5"
              />
              RUN_AGAIN
            </button>
          </header>
          <section
            className="mt-20"
          >
            <StatusBox
              indicator="10%"
              text="Running"
              title="PRODUCTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    PRODUCTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Running
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  10%
                </article>
              </section>
            </StatusBox>
            <StatusBox
              indicator="0%"
              text="Running"
              title="CONSUMPTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    CONSUMPTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Running
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  0%
                </article>
              </section>
            </StatusBox>
          </section>
        </main>
      </SystemChecksACPV>
    </Provider>
  </Router>
</BrowserRouter>
`;

exports[`SystemChecksACPV component render correctly when SUCCEEDED 1`] = `
<BrowserRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <SystemChecksACPV>
        <main
          className="page-height system-checks pl-10 pr-10"
        >
          <header
            className="is-clearfix has-text"
          >
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-left is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-chevron-left has-text-primary is-size-5 mr-5 mt-5"
              />
              SYSTEM_CHECKS
            </button>
            <button
              className="button is-text is-paddingless is-borderless has-text-grey has-text-weight-bold is-pulled-right is-size-6"
              disabled={false}
              onClick={[Function]}
            >
              <i
                className="sp-update left has-text-primary is-size-5 mr-5 mt-5"
              />
              RUN_AGAIN
            </button>
          </header>
          <section
            className="mt-20"
          >
            <StatusBox
              indicator="100%"
              text="Succeeded"
              title="PRODUCTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    PRODUCTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Succeeded
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  100%
                </article>
              </section>
            </StatusBox>
            <StatusBox
              indicator="100%"
              text="Succeeded"
              title="CONSUMPTION_CTE"
            >
              <section
                className="boxwrap mb-10 mt-10 pl-20 pr-20 pt-20 pb-20"
              >
                <article
                  className="left"
                >
                  <p
                    className="has-text-white has-text-weight-bold"
                  >
                    CONSUMPTION_CTE
                  </p>
                  <p
                    className="mt-10"
                  >
                    Succeeded
                  </p>
                </article>
                <article
                  className="is-flex file level is-right is-size-2 has-text-white has-text-weight-bold"
                >
                  100%
                </article>
              </section>
            </StatusBox>
          </section>
          <section
            className="mt-10"
          >
            <section
              className="is-flex"
            >
              <button
                className="button is-primary is-outlined is-uppercase mt-20 is-fullwidth mr-5"
                onClick={[Function]}
              >
                GO_TO_LIVE_DATA
              </button>
              <button
                className="button is-primary is-uppercase mt-20 is-fullwidth ml-5 is-fullwidth"
                onClick={[Function]}
              >
                COMMISSION
              </button>
            </section>
          </section>
        </main>
      </SystemChecksACPV>
    </Provider>
  </Router>
</BrowserRouter>
`;
